Sources/SourceKitLSP/Swift/CodeActions/ConvertIntegerLiteral.swift

```diff
@@ -0,0 +1,76 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SwiftRefactor
+import SwiftSyntax
+
+// TODO: Make the type IntegerLiteralExprSyntax.Radix conform to CaseEnumerable
+// in swift-syntax.
+
+extension IntegerLiteralExprSyntax.Radix {
+  static var allCases: [Self] = [.binary, .octal, .decimal, .hex]
+}
+
+/// Syntactic code action provider to convert integer literals between
+/// different bases.
+struct ConvertIntegerLiteral: SyntaxCodeActionProvider {
```

Comment:
Could you add a test case for this, checking that the refactoring do actually show up when retrieving code actions? Ie. a test in https://github.com/apple/sourcekit-lsp/blob/main/Tests/SourceKitLSPTests/CodeActionTests.swift